utils/build/docker/cpp/parametric/main.cpp

```diff
@@ -0,0 +1,57 @@
+#include <datadog/cerr_logger.h>
+#include <datadog/tracer.h>
+#include <datadog/tracer_config.h>
+
+#include <grpc/grpc.h>
+#include <grpcpp/server_builder.h>
+
+#include <cstdlib>
+
+#include "scheduler.h"
+#include "tracing_service.h"
+
+int main() {
+  // Might as well use a CerrLogger here, since we don't have our own logging library to adapt to the tracer with.
+  auto logger = std::make_shared<datadog::tracing::CerrLogger>();
+
+  // An event scheduler needs to be shared between the TracingService and the tracer.
+  auto event_scheduler = std::make_shared<ManualScheduler>();
+
+  // Populate tracer configuration with our objects and values.
+  datadog::tracing::TracerConfig config;
+  config.logger = logger;
+  config.agent.event_scheduler = event_scheduler;
+  config.defaults.service = "cpp-parametric-test";
+  config.defaults.environment = "staging";
+  config.defaults.name = "grpc.request";
+
+  // Finalize configuration so we can create a tracer.
+  auto finalized_config = datadog::tracing::finalize_config(config);
+  if (!finalized_config) {
+    logger->log_error("unable to initialize tracer:");
+    logger->log_error(finalized_config.error());
+    return 1;
+  }
+
+  auto tracer = std::make_unique<datadog::tracing::Tracer>(*finalized_config);
+
+  TracingService tracing_service(logger, std::move(tracer), event_scheduler);
```

Comment:
`unique_ptr` is a fine choice, but note that the following would also work:
```c++
TracingService tracing_service(logger, datadog::tracing::Tracer(*finalized_config), event_scheduler);
```
You could treat `Tracer` as a move-only type, or you could just copy it. `Tracer` is just a bag of values (mostly `shared_ptr`s). Its lifetime and multiplicity are irrelevant.